### DISCUSSION TITLE
가독성이 높은 코드를 작성하기 위해 어떤 노력을 하는지?

### DISCUSSION THREAD

**Discussion by GIVEN53** [2024-woowacourse-study/level-interview#23] [Clean Code]
.

**Comment by kunsanglee** [comment]
- 명명 규칙을 일관적으로 적용하기: 변수, 함수, 클래스 이름을 지을 때는 그 목적이나 역할을 명확하게 반영하는 이름을 사용합니다. 또한 프로젝트 내에서 일관된 명명 규칙을 적용하여 코드의 일관성을 유지합니다.

- 함수와 클래스의 크기를 적절하게 유지하기: 너무 긴 함수나 클래스는 이해하기 어렵습니다. 하나의 함수나 클래스가 하나의 기능만 수행하도록 작성하여 크기를 적절하게 유지합니다.

- 주석과 문서화: 코드에 주석을 추가하여 복잡한 로직이나 중요한 결정 사항을 설명합니다. 또한, API 문서화 도구를 사용하여 프로젝트의 문서를 작성하고 유지합니다.

- 코드의 구조를 명확하게 하기: 코드를 논리적인 단위로 구분하고, 적절한 들여쓰기와 공백을 사용하여 가독성을 높입니다. 또한, 코드의 흐름을 따라가기 쉽도록 구조를 명확하게 합니다.

- 매직 넘버(Magic Number) 사용을 피하기: 코드 내에서 직접적으로 숫자를 사용하는 대신, 그 의미를 명확히 할 수 있는 상수를 정의하여 사용합니다.

- 리팩토링: 기존의 코드를 주기적으로 검토하고, 더 나은 방식으로 개선하는 리팩토링 작업을 수행합니다. 이는 코드의 가독성을 높이고, 유지보수를 용이하게 합니다.

- 코드 리뷰: 팀원 간에 코드 리뷰를 정기적으로 수행하여, 다른 사람이 작성한 코드의 가독성을 평가하고, 개선점을 제시합니다.

**Comment by seunghye218** [comment]
저는 변수, 메서드, 클래스에 대해 명확하고 설명이 포함된 이름을 사용하기 위해 많은 노력을 기울였습니다. 줄임말이나 난해한 이름을 사용하지 않고 코드의 목적과 의도를 전달하는 이름을 선택합니다.

일관된 들여쓰기, 간격 및 명명 규칙을 사용하여 잘 정의된 코드 형식 스타일을 엄격하게 하려고 노력합니다. 이렇게 하면 코드 구조를 읽는 사람이 이해하는데 도움이 될 수 있습니다.

최대한 간결하고 표현력이 잘되는 코드를 작성하기 위해 노력합니다. 불필요한 복잡성, 중복성, 장황함을 피하여 코드가 목적을 명확하게 전달하도록 합니다.

**Comment by le2sky** [comment]
가독성 높은 코드를 작성하기 위해 실천하는 목표 중 하나는 '코드로 설명하기'입니다.
제가 작성한 코드를 주석으로 설명하지 않고, PR로도 설명하지 않으려고 최대한 노력합니다.
이러한 목적을 달성하기 위해서는 코드 자체로 설명이 되도록 노력해야하는데요.
코드 자체로 설명되도록 작성하는 노력을 통해서 가독성 높은 코드를 만들 수 있다는 생각을 가지고 있습니다.

**Comment by GIVEN53** [comment]
주석없이 이해되는 코드
변수, 메서드, 클래스명 명확하게
매직넘버 상수화
컨벤션
제3자의 입장에서 읽어보기
